### ISSUE TITLE
could not determine kind of name for C.SOCKET

### ISSUE THREAD

**Issue by SakHam** [pebbe/zmq4#152]
I use this guide **https://github.com/edgexfoundry/edgex-go/blob/master/ZMQWindows.md** to install ZMQ on windows cause it needed for edgex-go installation. In step 6 when i run **go get -v -x github.com/pebbe/zmq4**
I get this message :
**# github.com/pebbe/zmq4
Users\Sak\go\src\github.com\pebbe\zmq4\socketget_windows.go:20:11: could not determine kind of name for C.SOCKET**

**Comment by sujunbo** [comment]
Hello, Have you solved the above problem? I'm having the same problem as you right now?

**Comment by SakHam** [comment]
Hello,
No i quited to install it on windows. I did in ubuntu Virtual Machine.

**Comment by sujunbo** [comment]
You can run on unbuntu, right? I cannot operate in centos system, which is also a similar problem to yours. C name type not recognized

**Comment by sujunbo** [comment]
Hello, I have found the file you mentioned. You can add #include <winsock2.h> to the socketget_windows.go file

**Comment by SakHam** [comment]
Yes i installed ZMQ on ubuntu, once. Thank you very much for your help. When i will need it again i will check it on windows.

**Comment by sujunbo** [comment]
> 是的，我一次在ubuntu上安装了ZMQ。非常感谢您的帮助。当我再次需要它时，我将在Windows上对其进行检查。

Hi, can you share an article on how to install libzmq on Windows /centos?I am currently unable to use zmq4 correctly to manipulate zeromq on its environment.

**Comment by rsdmike** [comment]
So this seems to be a problem with 4.3.3, but not with 4.3.2, I used https://ci.appveyor.com/project/zeromq/libzmq/builds/25882500/job/s2yhbnjrdouq5qnm/artifacts 
and commit https://github.com/zeromq/libzmq/tree/a84ffa12b2eb3569ced199660bac5ad128bff1f0 and didnt have this error message.

**Comment by ghts** [comment]
I confirm zeromq 4.3.3 in msys2 have this issue as rsdmike said 
and solved by `add #include <winsock2.h> to the socketget_windows.go file`. as sujunbo said.

Thanks you both for kind and helpful info.

**Comment by pebbe** [comment]
I added `#include <winsock2.h>` to `socketget_windows.go`

**Comment by ghts** [comment]
I confirm with zmq4 1.2.2 there is no more error when 'go get github.com/pebbe/zmq4' with zeromq 4.3.3 in msys2 64 on Windows 10 ver 20H2.

Thank you for quick bugfix release.
